src/kem/kyber/helper.rs

```diff
@@ -0,0 +1,54 @@
+#[cfg(not(hax))]
+#[macro_export]
+macro_rules! cloop {
+    (for ($i:ident, $chunk:ident) in $val:ident.$values:ident.chunks_exact($($chunk_size:expr),*).enumerate() $body:block) => {
+        for $i in 0..$val.$values.len() / ($($chunk_size)*) {
+            let $chunk = &$val.$values[$i*($($chunk_size)*) .. $i*($($chunk_size)*)+($($chunk_size)*)];
+            $body
+        }
+    };
+    (for ($i:ident, $chunk:ident) in $val:ident.chunks_exact($($chunk_size:expr),*).enumerate() $body:block) => {
+        for $i in 0..$val.len() / ($($chunk_size)*) {
+            let $chunk = &$val[$i*($($chunk_size)*) .. $i*($($chunk_size)*)+($($chunk_size)*)];
+            $body
+        }
+    };
+    (for ($i:ident, $chunk:ident) in $val:ident.iter().enumerate() $body:block) => {
+        for $i in 0..$val.len() {
+            let $chunk = &$val[$i];
+            $body
+        }
+    };
+    (for ($i:ident, $chunk:ident) in $val:ident.into_iter().enumerate() $body:block) => {
+        for $i in 0..$val.len() {
+            let $chunk = $val[$i];
```

Comment:
should this be $item instead of $chunk?

Reply:
No, this should be `$chunk`. Otherwise the loop has no way to accessing it. Adding another `$item` would require changing the syntax of the loop that goes into the macro. We don't want that.

Reply:
I mean that this variable should not be called $chunk, since it is an item of the enumerate, not a chunk.